gquant/dataframe_flow/taskGraph.py

```diff
@@ -414,5 +630,12 @@ def draw(self, show=None, fmt='png', show_ports=False):
 
             plt = Image(pdot_out)
             display(plt)
+        elif show in ('lab',):
+            from gquantlab.gquantmodel import GQuantWidget
+            widget = GQuantWidget()
+            widget.value = self.export_task_speclist()
+            widget.set_taskgraph(self)
+            self.__widget = widget
+            return widget
```

Comment:
I think this logic should change a bit. The `get_widget` should be widget creator or maybe create a separate method to create a widget. The draw for lab should display the widget.
```
    def get_widget(self):
        if self.__widget is None:
            from gquantlab.gquantmodel import GQuantWidget
            widget = GQuantWidget()
            widget.value = self.export_task_speclist()
            widget.set_taskgraph(self)
            self.__widget = widget

        return self.__widget

```

In the draw method:
```
        elif show in ('lab',):
            from IPython.display import display
            widget = self.get_widget()
            display(widget)
            return widget
```

Reply:
Agree on the `get_widget` being a widget creator.
I don't think we need to call that "display" function. It breaks the behavior of ipywidgets. 
if you do this
```python
import ipywidgets
a = ipywidgets.HTML('<h1>hello</h1>')
```
the output cell is empty as assignment operation has not returned value.
It only shows the content in the output cell when you call
```
ipywidgets.HTML('<h1>hello</h1>')
```

Reply:
I see. Maybe the draw method shouldn't return the widget for `show='lab'` option.

That would be more consistent with `show='ipynb'` option.

The lab widget can be retrieved via `get_widget` (widget is too generic, I think it should be renamed to be more descriptive, maybe `get_quantlab`).

The pdot part is also irrelevant for the lab option.
```
    def draw(self, show='lab', fmt='png', show_ports=False):
        if show in ('ipynb',):
            pdot = self.to_pydot(show_ports)
            pdot_out = pdot.create(format=fmt)
            from IPython.display import display
            if fmt in ('svg',):
                from IPython.display import SVG as Image  # @UnusedImport
            else:
                from IPython.display import Image  # @Reimport

            plt = Image(pdot_out)
            display(plt)
        elif show in ('lab',):
            from IPython.display import display
            widget = self.get_widget()
            display(widget)
        
```

That would be one less cell in Jupyter to have to display a gquantlab widget.

I'm not sure if above breaks the sync states between widget references in other cells. I did this out of curiosity and there was some unexpected sync behavior with outputs collector node.

Reply:
These changes make the `draw` behave consistently so that fixes this issue. I'll mark resolved.